### DISCUSSION TITLE
FIxing a dependency tree by adding a constituent to the children of a verb

### DISCUSSION THREAD

**Discussion by quim66** [explosion/spaCy#11110] [Help: Coding & Implementations]
Hi
In a dependency tree a constituent is not in the children of a verb. I am trying to fix this tree by adding this constituent as a child of the verb. How can I do it?

Thanks.

**Comment by polm** [comment]
You should be able to just set `tok.head` to modify the dependency tree, though weird things can happen if you add cycles or other irregularities. 

Can you give an example of the kind of sentence you have and how you want it to be? If you have one sentence, typically the ROOT would be a verb and everything should be under it in the tree.

**Comment by quim66** [comment]
I am adapting a script that takes the subject, verb and object in English sentences for Spanish sentences. In a sentence like 'El profesor es un buen hombre' (the teacher is a good man), 'hombre' (man) is the ROOT. I want 'es' (is) to be the ROOT and ['profesor', 'hombre'] to be the children of 'es', as in English.

Thank you

**Reply by polm** [reply]
Ah, I see what you mean. 

Unfortunately I don't think there's any simple way to do this - [UD guidelines](https://universaldependencies.org/v2/copula.html) indicate that the head should be a noun for predicate nominatives like this. So the English structure is the exception here, and I think it only happens because the English annotations predate UD.

You could probably write code to transform trees, but it would get pretty tricky.
